### PR TITLE
Employees Creation

### PR DESCRIPTION
**Pull request by bibas07**
Employee Creation done

### CODE REVIEW

File: src/helpers/BaseHelper/base.helper.ts [hunk 1]
```diff
@@ -611,87 +597,105 @@ export class BaseHelper {
...
-        const error = this._page.locator('span.label.text-error');
-        const errorCount = await error.count();
-        if (errorCount > 0) {
-            console.log(chalk.red(`Error ocurred: ${errorCount}`));
-            for (let i = 0; i < errorCount; i++) {
-                const errorMsg = await error.nth(i).textContent();
-                return errorMsg;
-            }
-        }
+    public async toastSuccess() {
```
Comment (rumeshudash) [inline src/helpers/BaseHelper/base.helper.ts]: Please make toast helper.

File: src/helpers/BaseHelper/base.helper.ts [hunk 2]
```diff
@@ -611,87 +597,105 @@ export class BaseHelper {
...
+        // }
+        // if (toastErrorCount > 0) {
+        //     for (let i = 0; i < toastErrorCount; i++) {
+        //         const errorMsg = await toastError.nth(i).textContent();
+        //         return errorMsg;
+        //     }
+        // }
+    }
+    public async clickBreadCrumbsLink(linkName: string) {
+        await this.locate(`//a[contains(text(),"${linkName}")]`).click();
```
Comment (rumeshudash) [inline src/helpers/BaseHelper/base.helper.ts]: it should go in breadcrumb helper

File: src/helpers/BaseHelper/employeeCreation.helper.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    async verifyAddEmployeeForm(fieldName) {
+        await this._page.waitForTimeout(2000);
+        const parentLocator = this._page.locator(
+            `//div[@role="dialog"]//span[text()="${fieldName}"]/parent::label/parent::div`
+        );
+        await expect(
+            parentLocator,
+            chalk.red(`Dialog ${fieldName} visibility`)
+        ).toBeVisible();
```
Comment (rumeshudash) [inline src/helpers/BaseHelper/employeeCreation.helper.ts]: It is best if you move it into dialog helper and use here.

File: src/helpers/DesignationHelper/designation.helper.ts [hunk 4]
```diff
@@ -150,6 +151,13 @@ export class CreateDesignationHelper extends BaseHelper {
             .click();
     }
 
+    async verifyEmptyField() {
+        const name_field = await this.locate('input', {
+            name: 'name',
+        })._locator.inputValue();
```
Comment (rumeshudash) [inline src/helpers/DesignationHelper/designation.helper.ts]: Please move it to form helper

File: src/helpers/EmplyeeCreationHelper/employeeCreation.helper.ts [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    async clickActionOption(options: string) {
+        const optionContainer = this.locate('div', { role: 'menu' })._locator;
+        await optionContainer.getByRole('menuitem', { name: options }).click();
+    }
+    async changeTab(tabName: string) {
+        await this._page
+            .getByRole('tab', { name: tabName, exact: true })
+            .click();
+
+        await this._page.waitForTimeout(2000);
```
Comment (rumeshudash) [inline src/helpers/EmplyeeCreationHelper/employeeCreation.helper.ts]: You can use `TabHelper` here

File: src/helpers/ExpenseHelper/savedExpense.helper.ts [hunk 6]
```diff
@@ -19,9 +19,9 @@ export class SavedExpenseCreation extends BaseHelper {
...
 
-    public async toastMessage() {
-        return await this._page.locator("//div[@role='status']").textContent();
-    }
+    // public async toastMessage() {
+    //     return await this._page.locator("//div[@role='status']").textContent();
+    // }
 
     public async clickTab(buttonName: string) {
         await this.click({ role: 'tab', name: buttonName });
```
Comment (rumeshudash) [inline src/helpers/ExpenseHelper/savedExpense.helper.ts]: You can use `TabHelper` here

**Review by rumeshudash — CHANGES_REQUESTED** [review]

File: src/helpers/BaseHelper/base.helper.ts (hunk 2 above)
Comment (bibas07) [reply src/helpers/BaseHelper/base.helper.ts]: done

File: src/helpers/BaseHelper/employeeCreation.helper.ts (hunk 3 above)
Comment (bibas07) [reply src/helpers/BaseHelper/employeeCreation.helper.ts]: done

File: src/helpers/DesignationHelper/designation.helper.ts (hunk 4 above)
Comment (bibas07) [reply src/helpers/DesignationHelper/designation.helper.ts]: done

File: src/helpers/EmplyeeCreationHelper/employeeCreation.helper.ts (hunk 5 above)
Comment (bibas07) [reply src/helpers/EmplyeeCreationHelper/employeeCreation.helper.ts]: done

File: src/helpers/ExpenseHelper/savedExpense.helper.ts (hunk 6 above)
Comment (bibas07) [reply src/helpers/ExpenseHelper/savedExpense.helper.ts]: done

File: src/helpers/BaseHelper/base.helper.ts (hunk 1 above)
Comment (bibas07) [reply src/helpers/BaseHelper/base.helper.ts]: done

**Review by rumeshudash — APPROVED** [review]

**Pull request merged**